packages/blockchain-tools/src/contracts/soroban/credits/src/tests/utils.rs

```diff
@@ -0,0 +1,306 @@
+        account_id: account_id,
+        balance: balance, // 1000 XLM in stroops
+        seq_num: xdr::SequenceNumber(1),
+        num_sub_entries: 0,
+        inflation_dest: None,
+        flags: 0,
+        home_domain: xdr::String32::default(),
+        thresholds: xdr::Thresholds([1, 0, 0, 0]),
+        signers: xdr::VecM::default(),
+        ext: xdr::AccountEntryExt::V0,
+      }),
+      ext: xdr::LedgerEntryExt::V0,
+    });
+      
+    // Add the entry to storage
+    storage.put(
+      &key,
+      &entry,
+      None,
+      env.host().as_budget(),
+    ).unwrap();
+    
+    Ok(())
+  }).unwrap();
+}
+
+pub fn setup_tokens<'a>(
+  e: &Env,
+  admin: &Address
+) -> (
+  token::Client<'a>,
+  mintable_token::Client<'a>,
+  mintable_token::Client<'a>,
+
+) {
+  let usdc  = mintable_token::Client::new(
+    &e,
+    &e.register_stellar_asset_contract_v2(admin.clone()).address()
+  );
+
+  let xlm_address = deploy_native_sac(&e);
+  let xlm = token::Client::new(&e, &xlm_address);
+
+  let carbonSac = mintable_token::Client::new(
+    &e,
+    &e.register_stellar_asset_contract_v2(admin.clone()).address()
+  );
+
+  (xlm, usdc, carbonSac)
+}
+
+pub fn create_credit_contract<'a>(
+  e: &Env,
+  admin: &Address,
+  initiative: &String,
+  provider: &Address,
+  vendor: &Address,
+  bucket: i128,
+  xlm: &Address,
+  usdc: &Address,
+  carbonSac: &Address,
+  sink: &Address,
+  soroswapRouter: &Address
+) -> CreditsClient<'a> {
+  info!("Creating credit contract...");
+
+  let contract_id = e.register(
+    Credits,
+    (
+      admin.clone(),
+      initiative.clone(),
+      provider.clone(),
+      vendor.clone(),
+      bucket,
+      xlm.clone(),
+      usdc.clone(),
+      carbonSac.clone(),
+      sink.clone(),
+      soroswapRouter.clone()
+    )
+  );
+  let contract_client = CreditsClient::new(e, &contract_id);
+  info!("Credit Contract created!");
+  contract_client
+}
+
+pub fn create_sink_successors(e: &Env) -> (Address, Address) {
+  let admin = Address::generate(e);
+  let carbon_sac = Address::generate(e);
+  let carbonsink_sac = Address::generate(e);
+
+  let first_sink_id = e.register(
+    sink_contract::WASM, 
+    (&admin, &carbon_sac, &carbonsink_sac)
+  );
+  let first_sink_client = sink_contract::Client::new(e, &first_sink_id);
+  let second_sink_id = e.register(
+    sink_contract::WASM, 
+    (&admin, &carbon_sac, &carbonsink_sac)
+  );
+  first_sink_client.set_contract_successor(&second_sink_id);
+
+  let second_sink_client = sink_contract::Client::new(e, &first_sink_id);
+  let third_sink_id = e.register(
+    sink_contract::WASM, 
+    (&admin, &carbon_sac, &carbonsink_sac)
+  );
+  second_sink_client.set_contract_successor(&third_sink_id);
+
+  (first_sink_id, third_sink_id)
+}
+
+pub fn setup_soroswap_env<'a>(
+  e: &Env,
+  admin: &Address,
+  xlm: &token::Client<'a>,
+  usdc: &mintable_token::Client<'a>,
+  carbonSac: &mintable_token::Client<'a>,
```

Comment:
I like clear names, and there is a signature change required to match the token setup.

```
  xlm_id: &Address,
  usdc_sac: &StellarAssetContract,
  carbon_sac: &StellarAssetContract,
```